packages/components/tabs/src/TabsList.tsx

```diff
@@ -0,0 +1,183 @@
+import { useResizeObserver } from './useResizeObserver'
+
+export interface TabsListProps extends Omit<RadixTabs.TabsListProps, 'children'> {
+  /**
+   * Change the component to the HTML tag or custom component of the only child. This will merge the original component props with the props of the supplied element/component and change the underlying DOM node.
+   * @default false
+   */
+  asChild?: boolean
+  /**
+   * When true, keyboard navigation will loop from last tab to first, and vice versa.
+   * @default false
+   */
+  loop?: boolean
+  children: ReactElement[]
+}
+
+export const TabsList = forwardRef<HTMLDivElement, TabsListProps>(
+  (
+    {
+      /**
+       * Default Radix Primitive values
+       * see https://www.radix-ui.com/docs/primitives/components/tabs#list
+       */
+      asChild = false,
+      loop = false,
+      children,
+      className,
+      ...rest
+    },
+    ref
+  ) => {
+    const wrapperRef = useRef(null)
+    const innerRef = useRef(null)
+    const listRef = ref || innerRef
+    const { orientation } = useTabsContext()
+
+    const { width } = useResizeObserver(wrapperRef)
+
+    const [withArrows, setWithArrows] = useState<
+      Record<'prev' | 'next', 'visible' | 'hidden' | 'disabled'>
+    >({ prev: 'hidden', next: 'hidden' })
+
+    useEffect(() => {
+      /**
+       * Show/hide arrows
+       */
+      /* istanbul ignore if -- @preserve */
+      if (!('current' in listRef) || !listRef.current) {
+        return
+      }
+
+      if (orientation !== 'horizontal') {
+        setWithArrows({ prev: 'hidden', next: 'hidden' })
+      } else {
+        setWithArrows({
+          prev: listRef.current.scrollWidth > listRef.current.clientWidth ? 'visible' : 'hidden',
+          next: listRef.current.scrollWidth > listRef.current.clientWidth ? 'visible' : 'hidden',
+        })
+      }
+    }, [orientation, listRef, width])
+
+    useEffect(() => {
+      /**
+       * Enable/disable arrows
+       */
+      /* istanbul ignore if -- @preserve */
+      if (!('current' in listRef) || !listRef.current || withArrows.prev === 'hidden' || loop) {
+        return
+      }
+
+      const toggleArrowsVisibility = (target: HTMLDivElement) => {
+        setWithArrows({
+          prev: target.scrollLeft > 0 ? 'visible' : 'disabled',
+          next:
+            target.scrollLeft + target.clientWidth < target.scrollWidth ? 'visible' : 'disabled',
+        })
+      }
+
+      const currentList = listRef.current
+
+      toggleArrowsVisibility(currentList)
+
+      currentList.addEventListener('scroll', ({ target }) =>
+        toggleArrowsVisibility(target as HTMLDivElement)
+      )
+
+      return () =>
+        currentList.removeEventListener('scroll', ({ target }) =>
+          toggleArrowsVisibility(target as HTMLDivElement)
+        )
+    }, [listRef, withArrows.prev, loop])
+
+    const handlePrevClick = useCallback(() => {
```

Comment:
Is it necessary to wrap this function with a `useCallback`? Does it involve heavy computation?